src/crimson/osd/pg.cc

```diff
@@ -661,7 +661,24 @@ seastar::future<Ref<MOSDOpReply>> PG::handle_failed_op(
   }, load_obc_ertr::assert_all{ "can't live with object state messed up" });
 }
 
-seastar::future<Ref<MOSDOpReply>> PG::do_osd_ops(
+crimson::errorator<crimson::ct_error::eagain>::future<> PG::rep_repair_primary_object(
+  Ref<MOSDOp> m,
+  const hobject_t& oid,
+  eversion_t& v) 
+{
+  ceph_assert(is_primary());
+  logger().debug("{}: {} peers osd.{}", __func__, oid, get_acting_recovery_backfill());
+  // Add object to PG's missing set if it isn't there already
+  if (!get_local_missing().is_missing(oid)) {
+    peering_state.force_object_missing(pg_whoami, oid, v);
+  }
+  // Send Pull&Push messages to OSD holding replicas and add object to the recovery map
+  auto [op, fut] = get_shard_services().start_operation<crimson::osd::UrgentRecovery>(
```

Comment:
not sure why you drop `fut` on the floor. i think we should resume/restart the client request once the object is recovered.